app/controllers/pieces_controller.rb

```diff
@@ -13,6 +13,9 @@ def update
 
     if @piece.my_turn? == false || my_piece? == false
       flash[:alert] = "It's not your turn, you barnacle!"
+    elsif @piece.is_obstructed?(piece_params[:new_x].to_i, piece_params[:new_y].to_i) || 
+        !@piece.valid_move?(piece_params[:new_x].to_i, piece_params[:new_y].to_i)
+      flash[:alert] = "Sorry, can't move there"
```

Comment:
shouldn't this be "sorry, you can't move there, you empty-headed animal food trough wiper"??? ;)